app/scripts/modules/kubernetes/src/v2/manifest/delete/delete.html

```diff
@@ -0,0 +1,52 @@
+<div modal-page class="confirmation-modal">
+  <task-monitor monitor="ctrl.taskMonitor"></task-monitor>
+  <form role="form" name="deleteForm">
+    <modal-close dismiss="$dismiss()"></modal-close>
+    <div class="modal-header">
+      <h3>Delete {{ctrl.command.name}} in {{ctrl.command.location}}</h3>
+    </div>
+    <div class="modal-body confirmation-modal">
+      <div class="form-horizontal">
+        <div class="form-group form-inline">
+          <div class="col-md-3 sm-label-right">
+            Cascading
+            <help-field key="kubernetes.manifest.delete.cascading"></help-field>
+          </div>
+          <div class="col-md-3">
+            <div class="input-group">
+              <input type="checkbox"
+                     ng-model="ctrl.command.options.cascading"/>
+            </div>
+          </div>
+        </div>
+        <div class="form-group form-inline">
+          <div class="col-md-3 sm-label-right">
+            Grace Period
+            <help-field key="kubernetes.manifest.delete.gracePeriod"></help-field>
+          </div>
+          <div class="col-md-4">
+            <div class="input-group">
+              <input type="number"
+                     class="form-control input-sm highlight-pristine"
+                     ng-model="ctrl.command.options.gracePeriodSeconds"
+                     min="0"/>
+              <span class="input-group-addon">second<span ng-if="ctrl.command.options.gracePeriodSeconds !== 1">s</span></span>
+            </div>
+          </div>
+        </div>
+      </div>
+      <task-reason command="ctrl.command"></task-reason>
+    </div>
+    <div class="modal-footer">
+      <user-verification account="ctrl.command.account" verification="ctrl.verification"></user-verification>
+      <button type="submit" ng-click="ctrl.delete()" style="display:none"></button> <!-- Allows form submission via enter keypress-->
+      <button class="btn btn-default" ng-click="ctrl.cancel()">Cancel</button>
+      <button type="submit"
+              class="btn btn-primary"
+              ng-click="ctrl.delete()"
+              ng-disabled="!ctrl.isValid() || !resizeForm.$valid">
```

Comment:
Should be `deleteForm.$invalid`, (corresponds to the `name` attribute on your form tag). Making that change should allow you to omit the >= 0 logic in `isValid`, since your Grace Period input has `min=“0”`

Reply:
Ah neat